dataland-frontend/src/components/forms/CreateSFDRDataset.vue

```diff
@@ -0,0 +1,330 @@
+import DateFormField from "@/components/forms/parts/fields/DateFormField.vue";
+import SingleSelectFormField from "@/components/forms/parts/fields/SingleSelectFormField.vue";
+import MultiSelectFormField from "@/components/forms/parts/fields/MultiSelectFormField.vue";
+import AddressFormField from "@/components/forms/parts/fields/AddressFormField.vue";
+import RadioButtonsFormField from "@/components/forms/parts/fields/RadioButtonsFormField.vue";
+import SubmitButton from "@/components/forms/parts/SubmitButton.vue";
+import SubmitSideBar from "@/components/forms/parts/SubmitSideBar.vue";
+import YesNoNaFormField from "@/components/forms/parts/fields/YesNoNaFormField.vue";
+import UploadReports from "@/components/forms/parts/UploadReports.vue";
+import DataPointFormField from "@/components/forms/parts/kpiSelection/DataPointFormField.vue";
+import PercentageFormField from "@/components/forms/parts/fields/PercentageFormField.vue";
+import ProductionSitesFormField from "@/components/forms/parts/fields/ProductionSitesFormField.vue";
+import { objectDropNull, ObjectType } from "@/utils/UpdateObjectUtils";
+import { smoothScroll } from "@/utils/SmoothScroll";
+import { DocumentToUpload, ReportToUpload, uploadFiles } from "@/utils/FileUploadUtils";
+import MostImportantProductsFormField from "@/components/forms/parts/fields/MostImportantProductsFormField.vue";
+import { Subcategory } from "@/utils/GenericFrameworkTypes";
+import ProcurementCategoriesFormField from "@/components/forms/parts/fields/ProcurementCategoriesFormField.vue";
+
+export default defineComponent({
+  setup() {
+    return {
+      getKeycloakPromise: inject<() => Promise<Keycloak>>("getKeycloakPromise"),
+    };
+  },
+  name: "CreateSfdrDataset",
+  components: {
+    SubmitButton,
+    SubmitSideBar,
+    UploadFormHeader,
+    SuccessMessage,
+    FailMessage,
+    FormKit,
+    Card,
+    PrimeButton,
+    Calendar,
+    YesNoFormField,
+    InputTextFormField,
+    FreeTextFormField,
+    NumberFormField,
+    DataPointFormField,
+    DateFormField,
+    SingleSelectFormField,
+    MultiSelectFormField,
+    NaceCodeFormField,
+    AddressFormField,
+    RadioButtonsFormField,
+    YesNoNaFormField,
+    PercentageFormField,
+    ProductionSitesFormField,
+    MostImportantProductsFormField,
+    ProcurementCategoriesFormField,
+    UploadReports,
+  },
+  directives: {
+    tooltip: Tooltip,
+  },
+  emits: ["datasetCreated"],
+  data() {
+    return {
+      formId: "createSFDRForm",
+      waitingForData: true,
+      dataDate: undefined as Date | undefined,
+      companyAssociatedSfdrData: {} as CompanyAssociatedDataSfdrData,
+      sfdrDataModel: sfdrDataModel,
+      route: useRoute(),
+      message: "",
+      smoothScroll: smoothScroll,
+      uploadSucceded: false,
+      postSfdrDataProcessed: false,
+      messageCounter: 0,
+      checkCustomInputs,
+      documents: [] as DocumentToUpload[],
```

Comment:
In `CreateLksgDataset" this is a map:
`documents: new Map() as Map<string, DocumentToUpload>,`

The key of that map is the field name associated with the document.
Why is it a list here, and a map in Lksg?

Isn't the logic the same?

Reply:
Looking at this...

Reply:
In short, slightly different logic.
In `CreateLksgDataset` we use `YesNoFormField`s which handle single files and so the mapping functionality is pretty simple, too. In `CreateSfdrDataset` we have one component that handles multiple files and also uses the `UploadReports` component which emits a different object shape. It is much easier then to handle/replace multiple reports.

Reply:
I leave this open to recheck with it once @MarcinStanczyk7 has created the base-component for CreateLksgDataset.vue and CreateSFDRDataset.vue

Reply:
I changed it to use Map(). Apart from minor differences, it is pretty much the same as `CreateLksgDataset.vue`

`documents: new Map() as Map<string, DocumentToUpload>`

Reply:
Closing this since the base-component won't be created.

The transition to a map is ok for now.  Maybe if I review the latest version of the code, I can say more about it.